docs/architecture-decisions/multiple-flag-set-support.md

```diff
@@ -0,0 +1,163 @@
+---
+status: proposed
+author: @alexandraoberaigner
+created: 2025-05-28
+updated: -
+---
+
+# Add support for dynamic usage of Flag Sets to `flagd`
+
+The goal of this decision document is to establish flag sets as a first class concept in `flagd`, and support the dynamic addition/update/removal of flag sets at runtime.
+
+## Background
+
+`flagd` is a language-agnostic feature flagging engine that forms a core part of the OpenFeature ecosystem.
+Flag configurations can be stored in different locations so called `sources`. These are specified at startup, e.g.:
+
+````shell
+flagd start \
+  --port 8013 \
+  --uri file:etc/flagd/my-flags-1.json \
+  --uri https://my-flags-2.com/flags
+````
+
+Currently, we see 2 limitations:
+
+1. Once `flagd` is running, it is not possible to dynamically add flag sources.
+2. Logical groups of flags (for example the flags belonging to a particular development team or org unit) are tightly coupled to sources, and clients (usually flagd providers) must be aware of the source in order to select it.
```

Comment:
```suggestion
The primary object here is to remove the coupling between sources and "logical" groups of flags, so that provider's aren't required to know their set of flags are sources from a file/http resource, etc, but could instead just supply a logical identifier for their flag set.
```